src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -721,7 +721,7 @@ ssize_t sockinfo_tcp::tx(const tx_call_t call_type, const struct iovec* p_iov, c
 
 	if (unlikely(is_dummy) && !check_dummy_send_conditions(flags, p_iov, sz_iov)) {
 		unlock_tcp_con();
-		errno = EIO;
+		errno = EOPNOTSUPP;
```

Comment:
I don't think we should replace the errno value to EOPNOTSUPP.
There are several cases which we can enter inside the if statement, most of them are temporary. For example if the window is full.
Maybe EIO is not the most accurate one, but it is better than EOPNOTSUPP.